nautobot/dcim/models/locations.py

```diff
@@ -87,15 +87,26 @@ def clean(self):
 
     @property
     def display(self):
-        # prefix with parent name allows location type name identity if there are the same name exists under different parents.
-        display_str = self.name
-        parent = self.parent
-        for _ in range(10):  # "10 levels iteration"
-            if parent is None:
-                break
-            display_str = f"{parent.name} → {display_str}"
-            parent = parent.parent
-        return display_str
+        """
+        Prefix with parent name allows location type name identity if there are the same name exists under different parents.
+        self.ancestors() return all the preceding nodes from top down. So if we are looking at node C and its node structure is the following:
+            A
+           /
+          B
+         /
+        C
+        This method will return \"A → B → C\".
+        If self.ancestor() throws an ObjectDoesNotExist Exception during bulk operations, we handle it by only returning the name of the current node.
+        """
+        display_str = ""
+        try:
+            for ancestor in self.ancestors():
+                display_str += ancestor.name
+                display_str += " → "
+            display_str += self.name
+            return display_str
+        except ObjectDoesNotExist:
+            return self.name
```

Comment:
I still think this is incorrect because it loses potential ancestors that did exist.

What about (Psudocode-ish):

``` py
display_str = ""
member = self
while member is not None:
  try:
    display_str = f"{member.name} → {display_str}"
    member = member.parent
  except ObjectDoesNotExist:
    member = None

return display_str[:len(display_str)-3]
```

Reply:
Or perhaps move the 
Self.name and return display_str to the finally block:
'''
try:
   for ancestor in ancestors():
       display_str += ancestor.name + "->"

except ...

finally:
    return display_str += self.name
'''